apps/openmw/mwrender/npcanimation.cpp

```diff
@@ -959,6 +959,44 @@ void NpcAnimation::updateHolsteredWeapon()
                 addOrReplaceIndividualPart(weaptype, -1, 1, mesh, !weapon->getClass().getEnchantment(*weapon).empty(), &glowColor);
         }
     }
+
+    MWWorld::ConstContainerStoreIterator ammo = inv.getSlot(MWWorld::InventoryStore::Slot_Ammunition);
+    if (ammo != inv.end() && ammo.getType() == MWWorld::ContainerStore::Type_Weapon)
+    {
+        if (weapon == inv.end()) return;
+
+        const std::string &type = weapon->getClass().getTypeName();
+
+        // Show quiver only if the ranged weapon with the suitable ammo is equipped.
```

Comment:
IMO, it should show the quiver as long as the ammo is equipped, regardless of if you have a bow/crossbow equipped or not.

Reply:
No, you need the bow/crossbow equipped (line 980).
Also there is no much review for now - the code is in the early stage, and I even do not know if my approach is acceptable.

Reply:
> No, you need the bow/crossbow equipped (line 980).

That's what I mean. In my opinion, it shouldn't bother with that check and always show the quiver as long as the ammo is equipped.

Reply:
> That's what I mean. In my opinion, it shouldn't bother with that check and always show the quiver as long as the ammo is equipped.

We tried that, but quiver collides with two-handed weapon, so we decided to add this check, as in Skyrim, for now.

Reply:
Skyrim doesn't have such check. Quiver is always shown if ammo is equipped.
![Screenshot](https://user-images.githubusercontent.com/21265616/36643805-9a1f0b6c-1a61-11e8-964c-ed4aaa5d596c.png)